cmd/controller/app/controller.go

```diff
@@ -163,6 +168,24 @@ func buildControllerContext(ctx context.Context, stopCh <-chan struct{}, opts *o
 		return nil, nil, fmt.Errorf("error creating kubernetes client: %s", err.Error())
 	}
 
+	// The user may have enabled the gateway-shim controller but forgotten to
+	// install the Gateway API CRDs. Failing here will cause cert-manager to go
+	// into CrashLoopBackoff which is nice and obvious.
+	d := cl.Discovery()
+	resources, err := d.ServerResourcesForGroupVersion(gwapi.GroupVersion.String())
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't discover Gateway API resources (are the Gateway API CRDs installed?): %w", err)
+	}
+	if len(resources.APIResources) == 0 {
+		return nil, nil, fmt.Errorf("no gateway API resources were discovered (are the Gateway API CRDs installed?)")
+	}
```

Comment:
TBD: only do this when the `gateway-shim` controller has been added by the user (since it is not enabled by default).